perlock-core/src/main/java/name/mitterdorfer/perlock/SinglePathChangeListener.java

```diff
@@ -0,0 +1,18 @@
+package name.mitterdorfer.perlock;
+
+import java.nio.file.Path;
+import java.nio.file.WatchEvent;
+
+/**
+ * Callback interface for notifying the user when the chosen file changes.
+ * Suited for use as a functional interface
+ *
+ * <p>Implementation note: A <code>PathChangeListener</code> will be called from the internal <code>PathWatcher</code>
+ * thread. This implies that callback methods should return reasonably fast and offload heavy lifting to a dedicated
+ * thread. Otherwise, events may be lost.</p>
+ */
+public interface SinglePathChangeListener {
```

Comment:
So I will modify the `PathChangeListener` to have the method
```java
void onPathChanged(WatchEvent.Kind<Path> eventKind, Path path);
```
This is a hybrid between the two interfaces, as the `SingleChangePathListener` does not provide the `Path` since it is implicit.